### PR TITLE
Fixes #32607 - drop host from x-forwarded-for

### PR DESCRIPTION
**Pull request by lzap**
We do not use this at all.

### CODE REVIEW

File: modules/templates/proxy_request.rb [hunk 1]
```diff
@@ -36,7 +35,7 @@ def call_template(method, path, env, params, body = '')
       path = path.map { |x| CGI.escape(x) }.join('/')
       logger.debug "Template: request for #{path} using #{opts.inspect} at #{uri.host}"
       proxy_headers = extract_request_headers(env)
-      proxy_headers["X-Forwarded-For"] = "#{env['REMOTE_ADDR']}, #{proxy_ip}"
+      proxy_headers["X-Forwarded-For"] = "#{env['REMOTE_ADDR']}"
```
Comment (ekohl) [inline modules/templates/proxy_request.rb]:
```suggestion
      proxy_headers["X-Forwarded-For"] = env['REMOTE_ADDR']
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Last Friday I tried to read up on this and it's my reading that this is correct, just a minor inline code suggestion.

It would be even more correct to see if there already was a `X-Forwarded-For` header. If so, send that together with `REMOTE_ADDR`. Respecting `X-Forwarded-For` could be considered a feature though.

This might need a cherry pick into stable branches. If so, I'd limit the cherry pick to just what this PR is (the bugfix) and open a new PR for the feature.

**Comment by lzap** [comment]
Yeah sorry, rebased and amended the change. Since this is an important bug, I want just to fix this.

Introducing passing already-forwarded entries might actually trigger the bug we have in Foreman too, so that needs probably more work.

**Review by ekohl — APPROVED** [review]
Thanks!

**Review by ekohl — CHANGES_REQUESTED** [review]
Apologies, test failures are related.

**Comment by lzap** [comment]
For the record I asked one of our users to test this patch in https://community.theforeman.org/t/foreman-2-4-katello-4-ipxe-not-working/23398/45

**Comment by Thulium-Drake** [comment]
As also mentioned on the forum, this patch in combination with an edit of the ```foreman_proxies``` works like a charm:

```
# /etc/foreman/settings.yaml
# ... existing config
:trusted_proxies:
  - "127.0.0.0/8"
  - "::1"
  - "<IP address of the foreman server>"
```

**Review by Thulium-Drake — APPROVED** [review]

**Comment by ekohl** [comment]
> Apologies, test failures are related.

This is still true. The actual change looks good, but I don't want to break the build.

**Comment by lzap** [comment]
Fixed tests, rebased.

**Review by ekohl — APPROVED** [review]

**Comment by Thulium-Drake** [comment]
One addition, you also need to put _all_ smart proxy IPs on the same ```trusted_proxies``` list, if you don't you get 500 errors on the client. And 409 conflicts on the Foreman server (not the proxy)

**Pull request merged**